cmd/edit/machinepool/nodepool.go

```diff
@@ -307,13 +307,27 @@ func editAutoscaling(nodePool *cmv1.NodePool, minReplicas int, maxReplicas int)
 	asBuilder := cmv1.NewNodePoolAutoscaling()
 	changed := false
 
-	if nodePool.Autoscaling().MinReplica() != minReplicas && minReplicas >= 1 {
-		asBuilder = asBuilder.MinReplica(minReplicas)
-		changed = true
+	existingMinReplica := nodePool.Autoscaling().MinReplica()
+	existingMaxReplica := nodePool.Autoscaling().MaxReplica()
+
+	var min, max int
+	if minReplicas == 0 {
+		min = existingMinReplica
+	} else {
+		min = minReplicas
 	}
-	if nodePool.Autoscaling().MaxReplica() != maxReplicas && maxReplicas >= 1 {
-		asBuilder = asBuilder.MaxReplica(maxReplicas)
-		changed = true
+
+	if maxReplicas == 0 {
+		max = existingMaxReplica
+	} else {
+		max = maxReplicas
+	}
+
+	if existingMinReplica != minReplicas || existingMaxReplica != maxReplicas {
+		if min >= 1 && max >= 1 {
+			asBuilder = asBuilder.MinReplica(min).MaxReplica(max)
+			changed = true
+		}
```

Comment:
```suggestion
		if min >= 1 && max >= 1 {
			return cmv1.NewNodePoolAutoscaling().MinReplica(min).MaxReplica(max)
		}
```

the `asBuilder` is returned only if the code enters this path. We can return it directly here and get rid of the `asBuilder` and `changed` variables declared at the start of the method.

Reply:
Thanks @ziccardi , will make the change